### PR TITLE
fix: "no-dangle-underscore", and other nits.

### PR DESCRIPTION
**Pull request by eranhirsch**
And follow up with going over new rules and old rules and making sure our config is top-notch.

### CODE REVIEW

**Comment by eranhirsch** [comment]
This will have to wait for typescript-eslint to also update to v9, in the meantime this PR can't build in CI.

File: eslint.config.js [hunk 1]
```diff
@@ -135,12 +122,50 @@ export default config(
...
+      "prefer-spread": "error",
+      "prefer-template": "error",
+      radix: "error",
+      "require-unicode-regexp": "warn",
+      strict: "error",
+      "symbol-description": "warn",
+      "unicode-bom": ["error", "never"],
+      "vars-on-top": "warn",
+
+      // Code Quality "Params"
```
Comment (cjquines) [inline eslint.config.js:155]: **thought:** i am not usually a fan of these kinds of rules, but these are loose enough that i'm fine with this

File: src/pipe.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable complexity, max-depth, max-lines, max-statements -- TODO: We should strive to simplify the pipe implementation! */
```
Comment (cjquines) [inline src/pipe.ts:1]: **thought:** see, if we used iterators to deal with laziness, i think we can cut a lot of the complexity here

Comment (cjquines) [reply src/pipe.ts:1]: but of course, it would require a somewhat-substantial rewriting of things

File: src/sample.ts [hunk 3]
```diff
@@ -102,6 +102,7 @@ export function sample(...args: ReadonlyArray<unknown>): unknown {
   return purry(sampleImplementation, args);
 }
 
+// eslint-disable-next-line max-statements -- TODO: Can we simplify or split this?
```
Comment (cjquines) [inline src/sample.ts:105]: **question:** i see this function has a lot of guards, which we don't really have in other functions. for example, for `chunk`, we don't check that `size` is a positive integer. if we eliminate this, that'd bring us under the limit right?

**Review by cjquines — APPROVED** [review]

File: src/sample.ts (hunk 3 above)
Comment (eranhirsch) [reply src/sample.ts:105]: I think you are right and it also looks like how the built-in functions chose to handle this: for example `[1,2,3].slice(1.5)`.

I'll remove them

**Comment by eranhirsch** [comment]
It's taking the community longer than I thought to migrate to eslint v9, and the resulting changes in this PR don't need v9 to work - so we can ship the changes to the config itself later.

This mainly removes the underscore-based private functions in favor of proper names and directories.

Also removes runtime protections from sample which was the only function we had with runtime protections.

**Pull request merged**